core/vdbe/execute.rs

```diff
@@ -1810,11 +1810,17 @@ pub fn op_row_id(
             let rowid = {
                 let mut index_cursor = state.get_cursor(index_cursor_id);
                 let index_cursor = index_cursor.as_btree_mut();
-                index_cursor.rowid()?
+                let record = index_cursor.record();
+                let record = record.as_ref().unwrap();
+                let rowid = record.get_values().last().unwrap();
+                match rowid {
+                    RefValue::Integer(rowid) => *rowid as u64,
+                    _ => todo!(),
```

Comment:
this is not a `todo` right? 

why did we need this change from `index_cursor.rowid()?` overall?